meta-balena-common/recipes-containers/balena/balena/balena.service

```diff
@@ -2,14 +2,17 @@
 Description=Balena Application Container Engine
 Documentation=https://www.balena.io/docs/getting-started
 Wants=dnsmasq.service
-Requires=balena-engine.socket var-lib-docker.mount bind-etc-docker.service bind-home-root-.docker.service
-After=network.target balena-engine.socket var-lib-docker.mount bind-etc-docker.service bind-home-root-.docker.service dnsmasq.service rollback-altboot.service
+Requires=balena-engine.socket var-lib-docker.mount bind-etc-docker.service bind-home-root-.docker.service development-features.service
+After=network.target balena-engine.socket var-lib-docker.mount bind-etc-docker.service bind-home-root-.docker.service dnsmasq.service rollback-altboot.service development-features.service
+PartOf=development-features.target
 
 [Service]
 Type=notify
 Restart=always
 SyslogIdentifier=balenad
-ExecStart=/usr/bin/healthdog --healthcheck=/usr/lib/balena/balena-healthcheck /usr/bin/balenad --experimental --log-driver=journald -s @BALENA_STORAGE@ -H fd:// -H unix:///var/run/balena.sock -H unix:///var/run/balena-engine.sock --dns 10.114.102.1 --bip 10.114.101.1/24 --fixed-cidr=10.114.101.0/25 --max-download-attempts=10 --exec-opt native.cgroupdriver=systemd
+Environment="BALENA_DEVELOPMENT_ARGS="
+EnvironmentFile=-/var/volatile/development-features
+ExecStart=/usr/bin/healthdog --healthcheck=/usr/lib/balena/balena-healthcheck /usr/bin/balenad --experimental --log-driver=journald -s @BALENA_STORAGE@ -H fd:// -H unix:///var/run/balena.sock -H unix:///var/run/balena-engine.sock --dns 10.114.102.1 --bip 10.114.101.1/24 --fixed-cidr=10.114.101.0/25 --max-download-attempts=10 --exec-opt native.cgroupdriver=systemd $BALENA_DEVELOPMENT_ARGS
```

Comment:
This will conflict with https://github.com/balena-os/meta-balena/pull/2175. I guess we will see which gets merged first!